recipes/qt/6.x.x/conanfile.py

```diff
@@ -522,26 +526,133 @@ def package_id(self):
+            self.cpp_info.components[module].includedirs = ["include", os.path.join("include", "Qt%s" % module)]
+            self.cpp_info.components[module].defines = ["QT_%s_LIB" % module.upper()]
+            if module != "Core" and "Core" not in requires:
+                requires.append("Core")
+            self.cpp_info.components[module].requires = requires
+
+        def _create_plugin(pluginname, libname, type, requires):
+            assert pluginname not in self.cpp_info.components, "Plugin %s already present in self.cpp_info.components" % pluginname
+            self.cpp_info.components[pluginname].names["cmake_find_package"] = pluginname
+            self.cpp_info.components[pluginname].libs = [libname + libsuffix]
+            self.cpp_info.components[pluginname].libdirs = [os.path.join("res", "archdatadir", "plugins", type)]
+            self.cpp_info.components[pluginname].includedirs = []
+            if "Core" not in requires:
+                requires.append("Core")
+            self.cpp_info.components[pluginname].requires = requires
+
+        core_reqs = ["zlib::zlib"]
+        if self.options.with_pcre2:
+            core_reqs.append("pcre2::pcre2")
+        if self.options.with_doubleconversion:
+            core_reqs.append("double-conversion::double-conversion")
+        if self.options.get_safe("with_icu", False):
+            core_reqs.append("icu::icu")
+
+        _create_module("Core", core_reqs)
+        self.cpp_info.components["Core"].libs.append("Qt6Core_qobject%s" % libsuffix)
+        if self.options.gui:
+            gui_reqs = []
+            if self.options.with_freetype:
+                gui_reqs.append("freetype::freetype")
+            if self.options.with_libpng:
+                gui_reqs.append("libpng::libpng")
+            if self.options.get_safe("with_fontconfig", False):
+                gui_reqs.append("fontconfig::fontconfig")
+            if self.settings.os in ["Linux", "FreeBSD"]:
+                gui_reqs.append("xorg::xorg")
+                if not tools.cross_building(self, skip_x64_x86=True):
+                    gui_reqs.append("xkbcommon::xkbcommon")
+            if self.settings.os != "Windows" and self.options.get_safe("opengl", "no") != "no":
+                gui_reqs.append("opengl::opengl")
+            _create_module("Gui", gui_reqs)
+        if self.options.with_sqlite3:
+            _create_plugin("QSQLiteDriverPlugin", "qsqlite", "sqldrivers", ["sqlite3::sqlite3"])
+        if self.options.with_pq:
+            _create_plugin("QPSQLDriverPlugin", "qsqlpsql", "sqldrivers", ["libpq::libpq"])
+        if self.options.with_odbc:
+            if self.settings.os != "Windows":
+                _create_plugin("QODBCDriverPlugin", "qsqlodbc", "sqldrivers", ["odbc::odbc"])
+        _create_module("Network", ["openssl::openssl"] if self.options.openssl else [])
+        _create_module("PrintSupport", ["Gui", "Widgets"])
```

Comment:
Is widgets the one created below? https://github.com/conan-io/conan-center-index/pull/4763/files#diff-c548829da9d38ae15a8cb4447140bc50ac3f38250d1f348decd9569dc3298b24R594-R595

Reply:
yes